src/app/pages/quickMaps/pages/baselineDetails/mapView/mapView.component.ts

```diff
@@ -273,47 +384,107 @@ export class MapViewComponent implements AfterViewInit {
     this.thresholdLegend.addTo(this.thresholdMap);
   }
 
-
   private populateFeatureCollection(data: Array<SubRegionDataItem>): void {
     this.areaFeatureCollection = {
       type: 'FeatureCollection',
-      features: data
-        .map((item: SubRegionDataItem) => item.geoFeature)
-        .filter((item) => null != item),
+      features: data.map((item: SubRegionDataItem) => item.geoFeature).filter((item) => null != item),
     };
   }
 
-  private getColourAbsolute(absoluteValue: number): string {
-    return absoluteValue > 1500
-      ? '#2ca25f'
-      : absoluteValue > 1000
-        ? '#addd8e'
-        : absoluteValue > 500
-          ? '#ffeda0'
-          : absoluteValue > 250
-            ? '#feb24c'
-            : absoluteValue > 100
-              ? '#f03b20'
-              : absoluteValue > 50
-                ? '#bd0026'
-                : absoluteValue > 10
-                  ? '#7a0177'
-                  : '#354969';
-  }
-  private getColourThreshold(thresholdValue: number): string {
-    return thresholdValue > 99
-      ? '#187026'
-      : thresholdValue > 80
-        ? '#22f243'
-        : thresholdValue > 60
-          ? '#f0f01d'
-          : thresholdValue > 40
-            ? '#f08f11'
-            : thresholdValue > 20
-              ? '#e0071c'
-              : thresholdValue > 0
-                ? '#5e1d5c'
-                : '#0f0e0f';
+  private getColourRange(absoluteValue: number, thresholdValue: number, colourGradient: ColourGradientType): string {
+    if (colourGradient === ColourGradientType.REDYELLOWGREEN) {
+      return absoluteValue > 1500
```

Comment:
It would be better to separate out the threshold and absolute logic here.

This would be better if you used a switch.  I'm not sure if you've seen one like this, but you can arrange it in the following way, instead of the standard way:

```
switch(true) {
  case(thresholdValue > 99): return '#2ca25f';
  case(thresholdValue > 80): return '#addd8e';
  case(thresholdValue > 60): return '#ffeda0';
  case(thresholdValue > 40): return '#feb24c';
  ...
}
```

Or even better, have your array of thresholds and your array of colours then loop:
```
const thresholdColors = [
  {max: 0, color: '#2ca25f'},
  {max: 10, color: '#addd8e'},
  {max: 20, color: '#ffeda0'},
  ...
];
// simplified :-)
function getColour(value): string {
  const thresholdColor = this.thresholdColors.find(thisThresholdColor) => (value < thisThresholdColor.max);
  return (null != thresholdColor ) ? thresholdColor.color : '#123456';
}

```